### PR TITLE
lxc: Add support for creating profile from yaml (from Incus)

### PR DESCRIPTION
**Pull request by gabrielmougard**
cherry-picked from: https://github.com/lxc/incus/pull/756

### CODE REVIEW

**Comment by tomponline** [comment]
@gabrielmougard added "(from Incus)" to PR description to help with attribution, please can you do this for future PRs. Ta

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you add tests for this functionality.

File: lxc/profile.go [hunk 1]
```diff
@@ -319,19 +319,38 @@ func (c *cmdProfileCreate) command() *cobra.Command {
 	cmd.Short = i18n.G("Create profiles")
 	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
 		`Create profiles`))
+	cmd.Example = cli.FormatSection("", i18n.G(`incus profile create p1
```
Comment (tomponline) [inline lxc/profile.go:322]: You missed this bit

Comment (gabrielmougard) [reply lxc/profile.go:322]: oops thanks for that

Comment (gabrielmougard) [reply lxc/profile.go:322]: fixed

File: test/suites/basic.sh [hunk 2]
```diff
@@ -672,6 +672,29 @@ test_basic_usage() {
...
+EOF
+
+  lxc profile create foo < test_profile.yaml
+  lxc init testimage c1
+  lxc profile assign c1 foo
+  lxc config show c1 --expanded | grep -q "limits.cpu: \"2\""
+  lxc config show c1 --expanded | grep -q "limits.memory: 1024MB"
+  lxc delete -f c1
+  lxc profile delete foo
+  rm "test_profile.yaml"
```
Comment (tomponline) [inline test/suites/basic.sh:696]: Do we need the iterim file, shall we just print the profile text straight into the profile command?

Comment (gabrielmougard) [reply test/suites/basic.sh:696]: sure, let's do that. It's simpler after all.

File: test/suites/basic.sh [hunk 3]
```diff
@@ -672,6 +672,27 @@ test_basic_usage() {
...
+  limits.memory: 1024MB
+description: Test profile
+devices:
+  root:
+    path: /
+    pool: ${poolName}
+    type: disk
+EOF
+)
+  lxc init testimage c1
```
Comment (tomponline) [inline test/suites/basic.sh:689]: you can use `--profile` rather than needing to use `lxc profile assign`.

File: test/suites/basic.sh (hunk 2 above)
Comment (gabrielmougard) [reply test/suites/basic.sh]: fixed

File: test/suites/basic.sh [hunk 4]
```diff
@@ -672,6 +672,27 @@ test_basic_usage() {
...
+  lxc profile assign c1 foo
+  lxc config show c1 --expanded | grep -q "limits.cpu: \"2\""
```
Comment (tomponline) [inline test/suites/basic.sh:691]: Please can you use this style of explicit check https://github.com/canonical/lxd/commit/3d0ae50dd35539acdd1248f4ef50e746e7ff977e this is simpler and easier to debug when it fails in the tests as we can see the "wrong" value.

File: test/suites/basic.sh [hunk 5]
```diff
@@ -672,6 +672,27 @@ test_basic_usage() {
...
+  lxc config show c1 --expanded | grep -q "limits.memory: 1024MB"
```
Comment (tomponline) [inline test/suites/basic.sh:692]: Same comment as above, plus we tend to use MiB units in the tests.

File: test/suites/basic.sh (hunk 3 above)
Comment (gabrielmougard) [reply test/suites/basic.sh]: fixed

File: test/suites/basic.sh (hunk 4 above)
Comment (gabrielmougard) [reply test/suites/basic.sh]: fixed

File: test/suites/basic.sh [hunk 6]
```diff
@@ -672,6 +672,26 @@ test_basic_usage() {
...
+description: Test profile
+devices:
+  root:
+    path: /
+    pool: ${poolName}
+    type: disk
+EOF
+)
+  lxc init testimage c1 --profile foo
+  [ "$(lxc config show c1 --expanded | yq e '.config."limits.cpu"' -)" = "2" ]
```
Comment (tomponline) [inline test/suites/basic.sh:690]: We dont need another tool, we can should just use `lxc config get c1 limits.cpu --expanded`

File: test/suites/basic.sh [hunk 7]
```diff
@@ -672,6 +672,26 @@ test_basic_usage() {
...
+  [ "$(lxc config show c1 --expanded | yq e '.config."limits.memory"' -)" = "1024MiB" ]
```
Comment (tomponline) [inline test/suites/basic.sh:691]: We dont need another tool, we can should just use `lxc config get c1 limits.memory --expanded`

File: test/suites/basic.sh (hunk 6 above)
Comment (gabrielmougard) [reply test/suites/basic.sh]: fixed

File: test/suites/basic.sh (hunk 7 above)
Comment (gabrielmougard) [reply test/suites/basic.sh]: fixed

**Review by tomponline — APPROVED** [review]

**Pull request merged**